crates/typst/src/model/outline.rs

```diff
@@ -500,7 +500,18 @@ impl Show for Packed<OutlineEntry> {
         };
 
         // The body text remains overridable.
+        seq.push(
+            TextElem::new(
+                match TextElem::dir_in(styles) {
+                    crate::layout::Dir::RTL => "\u{202B}",
+                    _ => "\u{202A}",
+                }
+                .into(),
+            )
+            .pack(),
+        );
         seq.push(self.body().clone().linked(Destination::Location(location)));
+        seq.push(TextElem::new("\u{202C}".into()).pack());
```

Comment:
I think this kind of isolation could be moved into a helper function, e.g. `crate::text::isolate`.

Reply:
I also wonder whether we want LRE or LRI. Difference:

```typ
#set text(lang: "he")
#let val = "זוהי כותרת ארוכה מאוד שתורגמה על ידי תוכנת מחשב ועל כן עשויה להיות שגיאות דקדוקיות"
\u{202B} #val Pocus \u{202C} #box(width: 1fr, repeat[.]) 1 \
\u{2066} #val Pocus \u{2069} #box(width: 1fr, repeat[.]) 1
```

See also: https://github.com/typst/typst/issues/3696#issuecomment-2115547827

Reply:
So, this leaves this as the only remaining question before merge.

Reply:
> ```typst
> #set text(lang: "he")
> #let val = "זוהי כותרת ארוכה מאוד שתורגמה על ידי תוכנת מחשב ועל כן עשויה להיות שגיאות דקדוקיות"
> \u{202B} #val Pocus \u{202C} #box(width: 1fr, repeat[.]) 1 \
> \u{2066} #val Pocus \u{2069} #box(width: 1fr, repeat[.]) 1
> ```

I think you mean `\u{2067}` (assuming you meant to put RLI instead of LRI). With that change, the two lines look identical.

Reply:
My bad. But the lines being the same in this case doesn't mean it's always the same, so the question remains.